### PR TITLE
[cppgraphqlgen] Fix --head for new branch/option

### PR DESCRIPTION
**Pull request by wravery**
**Describe the pull request**

- What does your PR fix?

I'm not ready to update `cppgraphqlgen` to the v3.4.1 release yet because it depends on a change in `pegtl` which hasn't been released as v3.1.1 yet. However, installing both of those with `--head` should work, and right now it doesn't work for `cppgraphqlgen`. I renamed the default branch from `master` to `main`, and there's a new `GRAPHQL_UPDATE_VERSION=OFF` `CMake` option which the portfile needs to specify. Using that option with a prior commit will just be ignored.

- Which triplets are supported/not supported? Have you updated the CI baseline?

No, as far as I know the CI baselines should all work.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

Yes.

### CODE REVIEW

File: ports/cppgraphqlgen/portfile.cmake [hunk 1]
```diff
@@ -3,13 +3,13 @@ vcpkg_from_github(
...
     SHA512 88fedd83b49ae25f2165cff8e3a5e3e46ed853312658924f08b3192c790e92583dc8b8f7a88f39ec17798bd7d986143a78633a89d3867400539c4d315015e29b
-    HEAD_REF master
+    HEAD_REF main
 )
 
 vcpkg_configure_cmake(
     SOURCE_PATH ${SOURCE_PATH}
     PREFER_NINJA
-    OPTIONS -DGRAPHQL_BUILD_TESTS=OFF -DGRAPHQL_UPDATE_SAMPLES=OFF -DGRAPHQL_INSTALL_CONFIGURATIONS=Release
+    OPTIONS -DGRAPHQL_BUILD_TESTS=OFF -DGRAPHQL_UPDATE_VERSION=OFF -DGRAPHQL_UPDATE_SAMPLES=OFF -DGRAPHQL_INSTALL_CONFIGURATIONS=Release
```
Comment (NancyLi1013) [inline ports/cppgraphqlgen/portfile.cmake:12]:
```suggestion
    OPTIONS 
        -DGRAPHQL_BUILD_TESTS=OFF 
        -DGRAPHQL_UPDATE_VERSION=OFF 
        -DGRAPHQL_UPDATE_SAMPLES=OFF 
        -DGRAPHQL_INSTALL_CONFIGURATIONS=Release
```

File: ports/cppgraphqlgen/portfile.cmake [hunk 2]
```diff
@@ -3,13 +3,13 @@ vcpkg_from_github(
...
     OPTIONS_RELEASE -DGRAPHQL_INSTALL_CMAKE_DIR=${CURRENT_PACKAGES_DIR}/share -DGRAPHQL_INSTALL_TOOLS_DIR=${CURRENT_PACKAGES_DIR}/tools
```
Comment (NancyLi1013) [inline ports/cppgraphqlgen/portfile.cmake:13]:
```suggestion
    OPTIONS_RELEASE 
        -DGRAPHQL_INSTALL_CMAKE_DIR=${CURRENT_PACKAGES_DIR}/share 
        -DGRAPHQL_INSTALL_TOOLS_DIR=${CURRENT_PACKAGES_DIR}/tools
```

File: ports/cppgraphqlgen/portfile.cmake [hunk 3]
```diff
@@ -3,13 +3,13 @@ vcpkg_from_github(
...
     OPTIONS_DEBUG -DGRAPHQL_INSTALL_CMAKE_DIR=${CURRENT_PACKAGES_DIR}/debug/share -DGRAPHQL_INSTALL_TOOLS_DIR=${CURRENT_PACKAGES_DIR}/debug/tools
```
Comment (NancyLi1013) [inline ports/cppgraphqlgen/portfile.cmake:14]:
```suggestion
    OPTIONS_DEBUG 
        -DGRAPHQL_INSTALL_CMAKE_DIR=${CURRENT_PACKAGES_DIR}/debug/share 
        -DGRAPHQL_INSTALL_TOOLS_DIR=${CURRENT_PACKAGES_DIR}/debug/tools
```

**Review by NancyLi1013 — COMMENTED** [review]
Could you please also bump version in CONTROL file?

**Comment by NancyLi1013** [comment]
Hi @wravery 

Thanks for your PR.

Have you test if `./vcpkg install cppgraphqlgen--head` works fine on your local?

**Comment by wravery** [comment]
It does require that you install `pegtl` with `- - head` as well. Otherwise it tries to use the submodule copy of it, and that doesn't work with vcpkg.

**Comment by NancyLi1013** [comment]
Thanks for your response @wravery.

Have installed successfully now after install `pegtl `with` --head`.

**Comment by wravery** [comment]
> Could you please also bump version in CONTROL file?

Since I'm not actually changing the commit, I just added `-1` to the date-based version. Let me know if there's a better version scheme do handle that situation.

File: ports/cppgraphqlgen/CONTROL [hunk 4]
```diff
@@ -1,5 +1,5 @@
 Source: cppgraphqlgen
-Version: 2020-12-07
+Version: 2020-12-07-1
```
Comment (NancyLi1013) [inline ports/cppgraphqlgen/CONTROL:2]:
```suggestion
Version: 2020-12-07
Port-Version: 1
```
Actually, we use `port-version` field in this case now.

Please see this doc https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/control-files.md#port-version

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
LGTM now, thanks.

**Pull request merged**